e_os.h

```diff
@@ -226,6 +226,7 @@ extern FILE *_imp___iob;
 #  else
 #   define DEFAULT_HOME  "C:"
 #  endif
+#  define write _write
```

Comment:
Do you know if this is applicable across all MSVC versions we (claim to) support?

Reply:
I had tried to find out, but then I saw that already for VS2008 the `_write()` function is defined, so I thought that it will likely work for all versions used these days.

Reply:
I'd suggest to move it to vicinity where it's needed, i.e. to mem.c. There is #ifdef _WIN32 just before function that uses it. One can even wonder if it would be [more] appropriate to #ifndef _WIN32 that whole if (md_tracefd) clause that calls write. Thing is that *that* kind of file descriptors if not inherited on Windows and there is no way you can assign meaningful value to md_tracefd. Actually I'd rather vote for latter.

Otherwise to answer Richard's question, yes, it's applicable to all versions. Though for reference it's not really done because it's deprecated, but in first hand to facilitate linking of the static library into arbitrary context, statically into application, wrapped into .dll, and with /M flag of user's choice.

Reply:
> there is no way you can assign meaningful value to md_tracefd.

Well, one can assign 1 or 2...

Reply:
>> there is no way you can assign meaningful value to md_tracefd.

> Well, one can assign 1 or 2...

Yet I'd still suggest to omit that if(md_tracefd) in its entirety.

Reply:
Following these thoughts, I reverted my change to `e_os.h`
and added `# ifndef WIN32` around the clause using `write()`.
(Also improved indentation of the `#` preprocessor directives in its vicinity).